matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/RoomAPI.kt

```diff
@@ -156,4 +161,24 @@ internal interface RoomAPI {
                        @Path("state_event_type") stateEventType: String,
                        @Path("state_key") stateKey: String,
                        @Body params: Map<String, String>): Call<Unit>
+
+    /**
+     * Join the given room.
+     *
+     * @param roomId  the room id
+     * @param params the request body
+     */
+    @POST(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/join")
+    fun join(@Path("roomId") roomId: String,
+             @Body params: Map<String, String>): Call<Unit>
+
+    /**
+     * Leave the given room.
+     *
+     * @param roomId  the room id
+     * @param params the request body
+     */
+    @POST(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/leave")
+    fun leave(@Path("roomId") roomId: String,
+              @Body params: Map<String, String>): Call<Unit>
```

Comment:
Is it possible to create dedicated data class for `params`? Current usage is empty HashMap...